### PR TITLE
refactor magnitude computation using speclite

### PR DESCRIPTION
**Pull request by ntessore**
## Description

Refactor the magnitude computation to use `speclite` internally.

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/galaxy/spectrum.py [hunk 1]
```diff
@@ -122,99 +122,135 @@ def dirichlet_coefficients(redshift, alpha0, alpha1, z1=1., weight=None):
...
-            obs_tx = np.interp(obs_lam, band_lam, b, left=0, right=0)
-
-            # compute magnitude contribution from flux [numerator of (2)]
-            mag_ab[i, j, :] = -2.5*np.log10(np.trapz(spec_intg*obs_tx, obs_lam))
-
-    # combine AB magnitude [all of (2)]
-    mag_ab += np.atleast_1d(m_offs)[:, np.newaxis]
+    # if interpolating, split the redshift range into `interpolate` bits
+    if interpolate:
+        redshift_ = np.quantile(redshift, np.linspace(0, 1, interpolate))
```
Comment (rrjbca) [inline skypy/galaxy/spectrum.py:201]:
```suggestion
        redshift_ = np.linspace(np.min(redshift), np.max(redshift), interpolate)
```
Given that galaxies are not uniform in redshift, I'm not sure which of these would work better in general?

Comment (ntessore) [reply skypy/galaxy/spectrum.py:201]: That is the reason I changed it to produce equally populated redshift slices.

File: skypy/galaxy/spectrum.py [hunk 2]
```diff
@@ -122,99 +122,135 @@ def dirichlet_coefficients(redshift, alpha0, alpha1, z1=1., weight=None):
...
 
     Returns
     -------
-    mag_ab : (nz, nb, ns) array_like
-        Absolute AB magnitudes.
+    mag_ab : ([nz,] [ns,] nf,) array_like
+        The AB magnitude of each redshift (if given), each spectrum (if not
+        combined), and each filter.
 
     References
```
Comment (ntessore) [inline skypy/galaxy/spectrum.py:177]:
```suggestion
        combined), and each filter.

    Warnings
    --------
    The :mod:`speclite` package must be installed to use this function.

    References
```

File: skypy/galaxy/spectrum.py (hunk 1 above)
Comment (rrjbca) [reply skypy/galaxy/spectrum.py]: We discussed this offline and concluded that both are reasonable approaches and the difference between them should be small. Accuracy can always be improved with more interpolation points.

File: skypy/galaxy/spectrum.py [hunk 3]
```diff
@@ -228,8 +268,8 @@ def magnitudes_from_templates(coefficients, templates, bandpass, redshift=None,
         Array of spectrum coefficients.
     templates : spectral_data
         Template spectra.
-    bandpass : spectral_data
-        Bandpass filters.
+    filters : (nf,) `~speclite.filters.FilterSequence`
+        Sequence of bandpass filters.
```
Comment (ntessore) [inline skypy/galaxy/spectrum.py:272]:
```suggestion
    filters : str or list of str
        Bandpass filter specification for `~speclite.filters.load_filters`.
```

File: skypy/galaxy/spectrum.py [hunk 4]
```diff
@@ -243,55 +283,33 @@ def magnitudes_from_templates(coefficients, templates, bandpass, redshift=None,
 
     Returns
     -------
-    mag_ab : (ng, nb) array_like
-        The absolute AB magnitude of each object.
+    mag_ab : (ng, nf) array_like
+        The absolute AB magnitude of each object in each filter.
```
Comment (ntessore) [inline skypy/galaxy/spectrum.py:287]:
```suggestion
    mag_ab : (ng, nf) array_like
        The absolute AB magnitude of each object in each filter, where ``nf`` is the number of loaded filters.
```

File: skypy/galaxy/spectrum.py [hunk 5]
```diff
@@ -303,7 +321,7 @@ def stellar_mass_from_reference_band(coefficients, templates, magnitudes, bandpa
         Emission spectra of the templates.
     magnitudes : (ng,) array_like
         The magnitudes to match in the reference bandpass.
-    bandpass : spectral_data
+    filter : `~speclite.filters.FilterResponse`
         A single reference bandpass filter.
```
Comment (ntessore) [inline skypy/galaxy/spectrum.py:325]:
```suggestion
    filter : str
        A single reference bandpass filter specification for
        `~speclite.filters.load_filters`.
```

**Review by rrjbca — APPROVED** [review]

**Pull request merged**